panic: runtime error: invalid memory address or nil pointer dereference (git_repository)

### Terraform Version
 ```
Terraform v0.12.26
+ provider.git v0.1.1
```

### Affected Resource(s)
Please list the resources as a list, for example:
- `github_repository` (Data source)

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
data "git_repository" "main" {
  path = path.root
}
```

### Debug Output
N/A

### Panic Output
```
panic: runtime error: invalid memory address or nil pointer dereference
2020-07-03T10:50:50.808+0200 [DEBUG] plugin.terraform-provider-git_v0.1.1.exe: [signal 0xc0000005 code=0x0 addr=0x0 pc=0xec472c]
2020-07-03T10:50:50.808+0200 [DEBUG] plugin.terraform-provider-git_v0.1.1.exe:
2020-07-03T10:50:50.808+0200 [DEBUG] plugin.terraform-provider-git_v0.1.1.exe: goroutine 36 [running]:
2020-07-03T10:50:50.808+0200 [DEBUG] plugin.terraform-provider-git_v0.1.1.exe: github.com/go-git/go-git/v5.(*Remote).Config(...)
2020-07-03T10:50:50.808+0200 [DEBUG] plugin.terraform-provider-git_v0.1.1.exe:  /home/runner/go/pkg/mod/github.com/go-git/go-git/v5@v5.1.0/remote.go:58
2020-07-03T10:50:50.808+0200 [DEBUG] plugin.terraform-provider-git_v0.1.1.exe: github.com/innovationnorway/terraform-provider-git/internal/git.getRemoteURL(0xc000202ea0, 0x0, 0x0)
2020-07-03T10:50:50.808+0200 [DEBUG] plugin.terraform-provider-git_v0.1.1.exe:  /home/runner/work/terraform-provider-git/terraform-provider-git/internal/git/git.go:50 +0x4c
2020-07-03T10:50:50.808+0200 [DEBUG] plugin.terraform-provider-git_v0.1.1.exe: github.com/innovationnorway/terraform-provider-git/internal/git.dataSourceGitRepositoryRead(0xc0002dc0e0, 0xf3f360, 0xc000232150, 0xc0002dc0e0, 0x0)
2020-07-03T10:50:50.808+0200 [DEBUG] plugin.terraform-provider-git_v0.1.1.exe:  /home/runner/work/terraform-provider-git/terraform-provider-git/internal/git/data_source_git_repository.go:88 +0x7b2
```

### Expected Behavior
Terraform should not crash.

### Actual Behavior
Terraform crashes (a "panic" in the Go runtime).

### Steps to Reproduce
1. `git init`
2. `git add main.tf`
3. `git commit -m "test"`
4. `terraform init`
5. `terraform apply`

### References
N/A

--- a/internal/git/data_source_git_repository_test.go
+++ b/internal/git/data_source_git_repository_test.go
@@ -34,6 +34,9 @@ func TestAccDataSourceGitRepository_new(t *testing.T) {
 		t.Fatal(err)
 	}
 	err = ioutil.WriteFile(filepath.Join(dir, "test.txt"), []byte("hello world!"), 0644)
+	if err != nil {
+		t.Fatal(err)
+	}
 	_, err = worktree.Add("test.txt")
 	if err != nil {
 		t.Fatal(err)
@@ -45,7 +48,13 @@ func TestAccDataSourceGitRepository_new(t *testing.T) {
 			When:  time.Now(),
 		},
 	})
+	if err != nil {
+		t.Fatal(err)
+	}
 	obj, err := repo.CommitObject(commit)
+	if err != nil {
+		t.Fatal(err)
+	}
 	hash := obj.Hash.String()
 
 	path := filepath.ToSlash(dir)

--- a/internal/git/data_source_git_repository_test.go
+++ b/internal/git/data_source_git_repository_test.go
@@ -6,10 +6,64 @@ import (
 	"os"
 	"path/filepath"
 	"testing"
+	"time"
 
+	"github.com/go-git/go-billy/v5/osfs"
 	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/cache"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/storage/filesystem"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
 )
+
+func TestAccDataSourceGitRepository_new(t *testing.T) {
+	dir, err := ioutil.TempDir("", "acctest-*")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+	fs := osfs.New(dir)
+	dot, _ := fs.Chroot("storage")
+	storage := filesystem.NewStorage(dot, cache.NewObjectLRUDefault())
+	repo, err := git.Init(storage, fs)
+	if err != nil {
+		t.Fatal(err)
+	}
+	worktree, err := repo.Worktree()
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = ioutil.WriteFile(filepath.Join(dir, "test.txt"), []byte("hello world!"), 0644)
+	_, err = worktree.Add("test.txt")
+	if err != nil {
+		t.Fatal(err)
+	}
+	commit, err := worktree.Commit("test commit", &git.CommitOptions{
+		Author: &object.Signature{
+			Name:  "Terraform User",
+			Email: "terraform@example.com",
+			When:  time.Now(),
+		},
+	})
+	obj, err := repo.CommitObject(commit)
+	hash := obj.Hash.String()
+
+	path := filepath.ToSlash(dir)
+
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGitRepositoryPathConfig(path),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.git_repository.test", "url", ""),
+					resource.TestCheckResourceAttr("data.git_repository.test", "branch", "master"),
+					resource.TestCheckResourceAttr("data.git_repository.test", "commit_sha", hash),
+				),
+			},
+		},
+	})
+}
 
 func TestAccDataSourceGitRepository_path(t *testing.T) {
 	url := "https://github.com/innovationnorway/terraform-git-module-acctest.git"

--- a/internal/git/git.go
+++ b/internal/git/git.go
@@ -47,7 +47,7 @@ func getRepo(r RepoParams) (*git.Repository, error) {
 func getRemoteURL(repo *git.Repository) string {
 	remoteURL := ""
 	remote, _ := repo.Remote(git.DefaultRemoteName)
-	if len(remote.Config().URLs) > 0 {
+	if remote != nil {
 		remoteURL = remote.Config().URLs[0]
 	}
 

